airflow/models/taskinstance.py

```diff
@@ -299,11 +299,14 @@ def clear_task_instances(
         )
         dag_run_state = DagRunState(dag_run_state)  # Validate the state value.
         for dr in drs:
-            dr.state = dag_run_state
-            dr.start_date = timezone.utcnow()
-            if dag_run_state == DagRunState.QUEUED:
-                dr.last_scheduling_decision = None
-                dr.start_date = None
+            # Otherwise task(s) cleared in the middle of the running DagRun.
+            # There is no need to reset anything
+            if dr.state in [State.SUCCESS, State.FAILED]:
+                dr.state = dag_run_state
+                dr.start_date = timezone.utcnow()
+                if dag_run_state == DagRunState.QUEUED:
```

Comment:
This code will never be reached. 

That said, this is a change in behaviour and it seems to me that it'll cause a lot of confusion for the scheduler and users

Reply:
> This code will never be reached

Why not? 

> it'll cause a lot of confusion for the scheduler and users

it is causing confusion for our users when start_date isn't actual initial start_date. And regarding scheduler - I've tested that on my local Airflow and it works. Could you please be more specific about how exactly it will break something?

Reply:
it should be logically fine with all concept because it's a TI code and wherever we are clearing TI we don't want to reset `start_date` of the dag run, I don't see any issues with scheduler or other parts of the project

Reply:
and definitely there is no sense to reset the state from running to `queued` and then again to `running`

Reply:
> Why not?

sorry, read that as dr.state

Reply:
Well, I can't reason out the implications. Maybe @ashb can help out in the review